src/main/java/com/google/codeu/servlets/ChartServlet.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.codeu.servlets;
+
+import java.util.Optional;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Message;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+public class ChartServlet extends HttpServlet {
+
+  private Datastore datastore;
+
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    super.init(config);
+    datastore = createDatastore();
+  }
+
+  /*
+   * Instantiates a connection to a datastore. In tests, override this with a
+   * method that returns a fake/mock datastore object.
+   */
+  protected Datastore createDatastore() {
+    return new Datastore();
+  }
+
+  /**
+   * Responds with a JSON representation of {@link Message} data for logged-in user and another user. Responds with
+   * an empty array if the user is not logged in or another user is not provided.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType("application/json");
+
+    List<Message> messages = datastore.getMessagesBetweenTwoUsers(null, null);
```

Comment:
Interfaces should generally disallow null. This usage also makes the implementation of this operation needlessly complex and confusing (it is especially weird that "messages between two users" would allow either user to be omitted). Instead, create a different `getAllMessages` operation for your intended use case, please.

Reply:
Created a new getAllMessages function to resolve this.
The original intent was that it would be a good idea to reuse the function that was already created, but I see the weirdness of a function accepting nulls and outputting something.